xarray/core/indexes.py

```diff
@@ -440,7 +441,7 @@ class Indexes(collections.abc.Mapping):
 
     __slots__ = ("_indexes",)
 
-    def __init__(self, indexes):
+    def __init__(self, indexes: Mapping[Any, pd.Index]) -> None:
```

Comment:
@benbovy `Indexes` is typed here - is this compatible with the refactor (i.e. the `pd.Index`)? And if not can you adapt it in #5636?

Reply:
Ah yes I forgot to update the type here. `Indexes` is returned by both `Dataset.indexes` (pandas indexes) and `Dataset.xindexes` (xarray indexes), so `Union[pd.Index, Index]` is better.

Reply:
```suggestion
    def __init__(self, indexes: Mapping[Any, Union[pd.Index, Index]]) -> None:
```